pkg/ccl/changefeedccl/encoder_json.go

```diff
@@ -195,3 +198,40 @@ func (e *jsonEncoder) EncodeResolvedTimestamp(
 	}
 	return gojson.Marshal(jsonEntries)
 }
+
+var placeholderCtx = eventContext{topic: "topic"}
+
+// EncodeAsJSONChangefeedWithFlags implements the crdb_internal.to_json_as_changefeed_with_flags
+// builtin.
+func EncodeAsJSONChangefeedWithFlags(r cdcevent.Row, flags ...string) ([]byte, error) {
+	optsMap := make(map[string]string, len(flags))
+	for _, f := range flags {
+		split := strings.SplitN(f, "=", 2)
+		k := split[0]
+		var v string
+		if len(split) == 2 {
+			v = split[1]
+		}
+		optsMap[k] = v
+	}
+	opts, err := changefeedbase.MakeStatementOptions(optsMap).GetEncodingOptions()
+	if err != nil {
+		return nil, err
+	}
+	e, err := makeJSONEncoder(opts, changefeedbase.Targets{})
+	if err != nil {
+		return nil, err
+	}
+	return e.EncodeValue(context.TODO(), placeholderCtx, r, cdcevent.Row{})
+
+}
+
+func init() {
+	utilccl.RegisterArgType(cdcevent.Row{}, types.AnyTuple, func(d tree.Datum) interface{} {
```

Comment:
I just don't see how this is more readable than registering builtin where you
explicitly say that the function accepts a DTuple, and that you convert that tuple into a row, and then execute conversion.